python/lsst/daf/butler/_butler.py

```diff
@@ -533,56 +533,52 @@ def _standardizeArgs(self, datasetRefOrType: Union[DatasetRef, DatasetType, str]
+        primary values.
+
+        This allows a user to include a dataId dict with keys of
+        ``exposure.day_obs`` and ``exposure.seq_num`` instead of giving
+        the integer exposure ID.  It also allows a string to be given
+        for a dimension value rather than the integer ID if that is more
+        convenient. For example, rather than having to specifyin the
+        detector with ``detector.full_name``, a string given for ``detector``
+        will be interpreted as the full name and converted to the integer
+        value.
+
+        Keyword arguments can also use strings for dimensions like detector
+        and exposure but python does not allow them to include ``.`` and
+        so the ``exposure.day_obs`` syntax can not be used in a keyword
+        argument.
 
         Parameters
         ----------
-        datasetRefOrType : `DatasetRef`, `DatasetType`, or `str`
-            When `DatasetRef` the `dataId` should be `None`.
-            Otherwise the `DatasetType` or name thereof.
-        dataId : `dict` or `DataCoordinate`, optional
-            A `dict` of `Dimension` link name, value pairs that label the
-            `DatasetRef` within a Collection. When `None`, a `DatasetRef`
-            should be provided as the first argument.
-        collections : Any, optional
-            Collections to be searched, overriding ``self.collections``.
-            Can be any of the types supported by the ``collections`` argument
-            to butler construction.
-        allowUnresolved : `bool`, optional
-            If `True`, return an unresolved `DatasetRef` if finding a resolved
-            one in the `Registry` fails.  Defaults to `False`.
+        dataId : `dict` or `DataCoordinate`
+            A `dict` of `Dimension` link name, value pairs that will label the
+            `DatasetRef` within a Collection.
+        datasetType : `DatasetType`
+            The dataset type associated with this dataId. Required to
+            determine the relevant dimensions.
         kwds
             Additional keyword arguments used to augment or construct a
             `DataId`.  See `DataId` parameters.
 
         Returns
         -------
-        ref : `DatasetRef`
-            A reference to the dataset identified by the given arguments.
-
-        Raises
-        ------
-        LookupError
-            Raised if no matching dataset exists in the `Registry` (and
-            ``allowUnresolved is False``).
-        ValueError
-            Raised if a resolved `DatasetRef` was passed as an input, but it
-            differs from the one found in the registry.
-        TypeError
-            Raised if no collections were provided.
+        dataId : `dict` or `DataCoordinate`
+            The, possibly rewritten, dataId. If given a `DataCoordinate` and
+            no keyword arguments, the orginal dataId will be returned
+            unchanged.
+        kwds : `dict`
+            Any unused keyword arguments.
```

Comment:
I've been told to switch to `kwargs` in new code, and I think we usually put the leading `**` in the docs now, too.

Reply:
Yes, this was your code moved around. Will change it.